tap_ringcentral/__init__.py

```diff
@@ -1,19 +1,69 @@
 #!/usr/bin/env python3
 
 import singer
+import sys
 
-import tap_framework
-from tap_framework.state import save_state
 import argparse
 import json
 
+from tap_ringcentral.discover import discover
+
 from tap_ringcentral.client import RingCentralClient
-from tap_ringcentral.streams import AVAILABLE_STREAMS
+from tap_ringcentral.streams import STREAMS
+from tap_ringcentral.streams.base import is_stream_selected
 
 LOGGER = singer.get_logger()  # noqa
 
 
-class RingCentralRunner(tap_framework.Runner):
+class RingCentralRunner:
+    def __init__(self, args, client):
+        self.config = args.config
+        self.state = args.state
+        self.catalog = args.catalog
+        self.client = client
+        self.available_streams = STREAMS.values()
+
+    def save_state(self, state):
+        if not state:
+            return
+        LOGGER.info('Updating state.')
+        singer.write_state(state)
+
+    def do_discover(self):
+        LOGGER.info("Starting discovery")
+        catalog = discover()
+        json.dump(catalog.to_dict(), sys.stdout, indent=2)
+        LOGGER.info("Finished discover")
+
+    def get_streams_to_replicate(self):
+        streams = []
+
+        if not self.catalog:
+            return streams
+
+        for stream_catalog in self.catalog.streams:
+            if not is_stream_selected(stream_catalog):
+                LOGGER.info("'{}' is not marked selected, skipping."
+                            .format(stream_catalog.stream))
+                continue
```

Comment:
Wouldn't this do the same job?

```suggestion
        for stream in catalog.get_selected_streams(self.state):
```

Reply:
Simplified